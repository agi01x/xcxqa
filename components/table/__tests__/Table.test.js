import React from 'react';
import { render, shallow, mount } from 'enzyme';
import Table from '..';

const { Column, ColumnGroup } = Table;

describe('Table', () => {
  it('renders JSX correctly', () => {
    const data = [{
      key: '1',
      firstName: 'John',
      lastName: 'Brown',
      age: 32,
    }, {
      key: '2',
      firstName: 'Jim',
      lastName: 'Green',
      age: 42,
    }];

    const wrapper = render(
      <Table dataSource={data} pagination={false}>
        <ColumnGroup title="Name">
          <Column
            title="First Name"
            dataIndex="firstName"
            key="firstName"
          />
          <Column
            title="Last Name"
            dataIndex="lastName"
            key="lastName"
          />
        </ColumnGroup>
        <Column
          title="Age"
          dataIndex="age"
          key="age"
        />
      </Table>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('updates columns when receiving props', () => {
    const columns = [{
      title: 'Name',
      key: 'name',
      dataIndex: 'name',
    }];
    const wrapper = shallow(<Table columns={columns} />);
    const newColumns = [{
      title: 'Title',
      key: 'title',
      dataIndex: 'title',
    }];
    wrapper.setProps({ columns: newColumns });

    expect(wrapper.instance().columns).toBe(newColumns);
  });

  it('loading with Spin', async () => {
    const loading = {
      spinning: false,
      delay: 500,
    };
    const wrapper = mount(<Table loading={loading} />);
    expect(wrapper.find('.ant-spin')).toHaveLength(0);
    expect(wrapper.find('.ant-table-placeholder').text()).not.toEqual('');

    loading.spinning = true;
    wrapper.setProps({ loading });
    expect(wrapper.find('.ant-spin')).toHaveLength(0);

    await new Promise(resolve => setTimeout(resolve, 500));
    wrapper.update();
    expect(wrapper.find('.ant-spin')).toHaveLength(1);
  });

  it('renders custom components correctly when it changes', () => {
    const BodyWrapper1 = props => <tbody id="wrapper1" {...props} />;
    const BodyWrapper2 = props => <tbody id="wrapper2" {...props} />;
    const wrapper = mount(<Table components={{ body: { wrapper: BodyWrapper1 } }} />);
    wrapper.setProps({ components: { body: { wrapper: BodyWrapper2 } } });
    expect(wrapper.find('tbody').props().id).toBe('wrapper2');
  });
});
