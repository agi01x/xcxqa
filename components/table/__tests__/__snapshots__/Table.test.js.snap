// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders JSX correctly 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table ant-table-default ant-table-scroll-position-left"
      >
        <div
          class="ant-table-content"
        >
          <div
            class="ant-table-body"
          >
            <table
              class=""
            >
              <colgroup>
                <col />
                <col />
                <col />
              </colgroup>
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class=""
                    colspan="2"
                  >
                    <div>
                      Name
                    </div>
                  </th>
                  <th
                    class=""
                    rowspan="2"
                  >
                    <div>
                      Age
                    </div>
                  </th>
                </tr>
                <tr>
                  <th
                    class=""
                  >
                    <div>
                      First Name
                    </div>
                  </th>
                  <th
                    class=""
                  >
                    <div>
                      Last Name
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="1"
                >
                  <td
                    class=""
                  >
                    <span
                      class="ant-table-row-indent indent-level-0"
                      style="padding-left:0px"
                    />
                    John
                  </td>
                  <td
                    class=""
                  >
                    Brown
                  </td>
                  <td
                    class=""
                  >
                    32
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="2"
                >
                  <td
                    class=""
                  >
                    <span
                      class="ant-table-row-indent indent-level-0"
                      style="padding-left:0px"
                    />
                    Jim
                  </td>
                  <td
                    class=""
                  >
                    Green
                  </td>
                  <td
                    class=""
                  >
                    42
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
